ui/recieve_page.go

```diff
@@ -1 +1,301 @@
+												})
+											}),
+											layout.Rigid(func() {
+												layout.Inset{Left: unit.Dp(22)}.Layout(win.gtx, func() {
+													win.outputs.selectedWalletBalLabel.Layout(win.gtx)
+												})
+											}),
+										)
+									})
+								}),
+							)
+						})
+					}
+					decredmaterial.Card{}.Layout(win.gtx, selectedDetails)
+				}),
+			)
+		}),
+	)
+}
+
+func (win *Window) qrCodeAddressColumn() {
+	qrCode, err := qrcode.New(win.addrs, qrcode.Highest)
+	if err != nil {
+		win.outputs.err.Text = err.Error()
+		return
+	}
+
+	qrCode.DisableBorder = true
+	layout.Flex{Axis: layout.Vertical}.Layout(win.gtx,
+		layout.Rigid(func() {
+			layout.Inset{Top: unit.Dp(16)}.Layout(win.gtx, func() {
+				layout.Flex{}.Layout(win.gtx,
+					layout.Flexed(0.2, func() {
+					}),
+					layout.Flexed(1, func() {
+						img := win.theme.Image(paint.NewImageOp(qrCode.Image(520)))
+						img.Src.Rect.Max.X = 521
+						img.Src.Rect.Max.Y = 521
+						img.Scale = 0.5
+						img.Layout(win.gtx)
+					}),
+				)
+			})
+		}),
+		layout.Rigid(func() {
+			layout.Inset{Top: unit.Dp(16)}.Layout(win.gtx, func() {
+				layout.Flex{}.Layout(win.gtx,
+					layout.Flexed(0.1, func() {
+					}),
+					layout.Flexed(1, func() {
+						win.receiveAddressColumn()
+					}),
+				)
+			})
+		}),
+	)
+}
+
+func (win *Window) receiveAddressColumn() {
+	layout.Flex{}.Layout(win.gtx,
+		layout.Rigid(func() {
+			win.outputs.receiveAddressLabel.Text = win.addrs
+			win.outputs.receiveAddressLabel.Layout(win.gtx)
+		}),
+		layout.Rigid(func() {
+			layout.Inset{Left: unit.Dp(16)}.Layout(win.gtx, func() {
+				for win.inputs.receiveIcons.copy.Clicked(win.gtx) {
+					clipboard.WriteAll(win.addrs)
+					win.addressCopiedLabel.Text = "Address Copied"
+					time.AfterFunc(time.Second*9, func() {
+						win.addressCopiedLabel.Text = ""
+					})
+				}
+				win.outputs.copy.Layout(win.gtx, &win.inputs.receiveIcons.copy)
+			})
+		}),
+	)
+}
+
+func (win *Window) setDefaultPageValues() {
+	info := win.walletInfo.Wallets[win.selected]
+
+	for i := range info.Accounts {
+		if len(info.Accounts) == 0 {
+			continue
+		}
+
+		for win.inputs.receiveIcons.newAddress.Clicked(win.gtx) {
+			addr, err := win.wallet.NextAddress(info.ID, info.Accounts[i].Number)
+			if err != nil {
+				win.outputs.err.Text = err.Error()
+				return
+			}
+			info.Accounts[i].CurrentAddress = addr
+			win.isNewAddrModal = false
+		}
+
+		win.setSelectedAccount(info, info.Accounts[i])
+		break
+	}
+}
+
+func (win *Window) setSelectedAccount(wallet wallet.InfoShort, account wallet.Account) {
+	win.selectedWallet = &wallet
+	win.selectedAccount = &account
+
+	win.outputs.selectedAccountNameLabel.Text = account.Name
+	win.outputs.selectedWalletNameLabel.Text = wallet.Name
+	win.outputs.selectedWalletBalLabel.Text = dcrutil.Amount(account.SpendableBalance).String()
+	win.outputs.selectedAccountBalanceLabel.Text = wallet.Balance
+	win.addrs = account.CurrentAddress
+}
```

Comment:
Remove. The selected account just needs to be an int field of wallet indicating the selected account's index in `InfoShort.Accounts`

Reply:
Are you referring to line 243 n 244?

Reply:
I'm referring to the whole function. The setting of the selected account should be handled by the a change in the account Select widget